src/taskgraph/transforms/from_deps.py

```diff
@@ -183,9 +183,10 @@ def from_deps(config, tasks):
                 )
 
             new_task = deepcopy(task)
-            new_task["dependencies"] = {
-                dep.kind if unique_kinds else dep.label: dep.label for dep in group
-            }
+            new_task.setdefault("dependencies", {})
+            new_task["dependencies"].update(
```

Comment:
nit: `setdefault` returns the dict too, so you could just chain `.update` onto the end of that function.